### PR TITLE
Add button component Issue #20

### PR DESCRIPTION
**Pull request by geemanjs**

### CODE REVIEW

**Comment by lennym** [comment]
This could use a `type` prop so it can be used to render a `<button type="submit" />` and not break good old-fashioned form submission.

Also, it might be good if you could pass an optional `href` that instead rendered a `<a href="" class="button">...</a>`

tl;dr - I don't want to have to rely on `onClick` to do things.

**Comment by geemanjs** [comment]
@lennym That should be added on already because of this on the Button tag.. 
```
{...props} 
```
You can basically pass through anything as it's probable people might want to add some aria attributes as extras as well. I will add type to the propTypes though.

As for the `<a>` tag a lot of the react world will use something like [react-router](https://reacttraining.com/react-router/web/guides/philosophy). So maybe having the option to pass what 'tag' you'd want it to be in. 

Take a look at this video https://www.youtube.com/watch?v=gNeavlJ7lNY it's pretty awesome

**Comment by geemanjs** [comment]
Let me know what you think to these changes!

File: src/button/index.jsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  return component;
+}
+
+export const Button = buttonOf("button");
+Button.propTypes = Object.assign({ type: PropTypes.string }, Button.propTypes);
+
+export const LinkButton = buttonOf("a");
+LinkButton.propTypes = Object.assign({ href: PropTypes.string.isRequired }, LinkButton.propTypes);
+
+export default Button;
```
Comment (lennym) [inline src/button/index.jsx]: I really like this, but the mixed use of double and singe quotes makes me twitch a little. I know it's wholly cosmetic, but could we standardise on single-quotes?

**Comment by lennym** [comment]
I like the new changes a lot. I know it's probably slightly unconventional, but all my use cases at the moment are SSR only, and I'm trying to only put any script on the client at all when it's absolutely necessary, so while I can use "old fashioned" constructs like links and form-submissions to do what I need, I prefer to.

Thanks a lot for humouring me.

File: src/button/index.jsx (hunk 1 above)
Comment (geemanjs) [reply src/button/index.jsx]: Should be fixed now! This is why I like tools like prettier!

**Pull request merged**